library/alloc/src/collections/btree/map.rs

```diff
@@ -460,7 +468,7 @@ impl<K: Debug + Ord, V: Debug> Debug for Entry<'_, K, V> {
 pub struct VacantEntry<'a, K: 'a, V: 'a> {
     key: K,
     handle: Handle<NodeRef<marker::Mut<'a>, K, V, marker::Leaf>, marker::Edge>,
-    length: &'a mut usize,
+    maplease: DormantMutRef<'a, BTreeMap<K, V>>,
```

Comment:
This is a strange field name... why "please"? Is this the verb ("to please") or more like "please do X"? And what's "ma"?

Maybe this was an at attempt at humor, but it was lost on me. ;) Not sure if field names in unsafe code are the right place for humor. Anyway the name should be explained. (Same for the other structs where it occurs.)

Reply:
It should have been `map_lease` and I don't know if "lease" is the right word here, for a kind of contract subletting a house that you're renting (borrowing) yourself.

Reply:
Oh d'oh, I didn't see that way of splitting the words at all.^^

Maybe `map_dormant`, to match the type name?

Reply:
I landed on
```
let (dormant_map, self2) = DormantMutRef::new(self);
```
for the time being. But I wonder about:
```
let (dormant_map, live_map) = DormantMutRef::new(self);
let (map, dormant_map) = Reborrow::new(self); // and awaken → pop
let (map, dormant_map) = StackedBorrow::new(self);
```

Reply:
I definitely prefer having the live reference in front and the dormant one at the back.

As to names, I can't find anything resembling the ugly suffix. There is `std::cell::Ref`, whatever that actually is, but the `ptr::NonNull` wrapper is hard to base naming on, because both the words "ptr" and "Null" already give away it's about a pointer, and "Dormant" doesn't reveal anything. "Reborrow" isn't great either because most the class is mostly about the pointer sleeping behind the reference that is a reborrow, I think.

Reply:
> std::cell::Ref, whatever that actually is,

It's a guard type for `RefCell`, similar to `RwLockReadGuard` for `RwLock`.

I like "dormant", TBH.^^